lib/queuePopulator/IngestionPopulator.js

```diff
@@ -275,10 +282,11 @@ class IngestionPopulator {
                 return ingestionReader;
             });
             this.logReadersUpdate.push(...newRaftReaders);
-            return cb();
+            return undefined;
         });
         this.log.info('waiting to be provisioned a log source',
                       { zkEndpoint });
+        return cb();
```

Comment:
This is odd, you are calling the callback outside of the `subscribe` call which is async...